### ISSUE TITLE
[BUG] WMI Query is still broken

### ISSUE THREAD

**Issue by JustArchi** [SteamRE/SteamKit#305]
See https://github.com/SteamRE/SteamKit/issues/254 for initial issue.

It seems that `LogOn()` is still broken for several machines without capabilities of executing WMI queries. Because it's not possible to overload the method or specify whether `GenerateMachineID()` should execute or not, I'm reopening the issue.

I think it would be best to simply silently ignore the exception and proceed with the rest instead of rethrowing exception back to consumer, it's not like there is much to be done anyway, but that'my thoughts.

Basically logging in on such machine is not possible without resorting to hacks and sending custom request which omits generation of `logon.Body.machine_id` field.

Thanks!

![:3](http://image.prntscr.com/image/bc388ae5eadf435bb43c636cf1536928.png)

**Comment by VoiDeD** [comment]
Ideally, WMI failures should be handled gracefully. I think we'd need to decide if the response is a null machine_id or fall back to the `DefaultInfoProvider` machine info provider.

**Comment by Netshroud** [comment]
Whatever we fall back to would be the implementation for .NET Core, since System.Management isn't available there.

**Comment by Netshroud** [comment]
This was resolved by 645b91a - we switched from WMI to Win32 `DeviceIoControl`.
